docs/development/pigs/pig-012.rst

```diff
@@ -0,0 +1,334 @@
+
+
+Alternatives
+============
+
+A different approach to this high-level interface API is that of command line tools executed from the 
+terminal, what  `Fermitools`_ and `ctools`_ do, where each tool is simple/atomic enough to allow users
+to inspect the output results before taking a decision on how to run and set the parameter values for
+the next tool. A similar approach could be done with the Gammapy high-level interface API, but inside
+a notebook or IPython session.
+
+Concerning the code implementation, `ctapipe tools`_ provides a solution based on Python traitlets,
+acting as an extensible framework to easily transform Python classes into command line tools.
+We will explore the adoption of this approach after Gammapy v1.0 since it requires a considerable
+effort on refactoring of the Gammapy code-base.
+
+Another config-file based solution is what is implemented in `Enrico`_. It performs basic orchestrated
+analysis workflows using a set of input parameters that the user provides via a configuration file. 
+The user may be guided in the declaration of values for the config file using an assistant command line
+tool for config-file building, which asks for parameter values providing also defaults. This is done in 
+`Enrico`_ with  ``enrico_config`` and ``enrico_xml``, where each workflow is set-up and then run with its 
+own command line tool. In our case, we define the workflow steps in a simple Python script and declare 
+parameter-value pairs in a configuration file. The Python script is then executed to run the workflow. 
+
+Also Python scripts and/or notebook files could be generated with an assistant command line tool. Then, 
+the user could edit and tweak the config files, scripts or notebooks. There isn't much precedence for this
+workflow in science, but a lot of dev-ops and programming tools work like that, it is a standard technique.
+One random example of such a tool is the `Angular CLI`_, or `cookiecutter`_. 
+
+Task list
+=========
+- Prototype for a manager class, agents, tools, etc.
+- Define a syntax for the declaration of parameter-value pairs needed for all tools in the analysis process.
+- Develop the session manager class responsible to drive the orchestration of tools in the analysis process.
+- Develop the tools classes responsible to perform each one of the tasks in the analysis process.
+- Develop the small set of helpers command line tools described above.
+- Design use cases and/or choose among the existing tutorials or benchmarks those that may be translated into high-level interface notebooks.
+- Provide notebooks using the high-level interface API for each of the chosen tutorials, benchmarks and/or use cases identified.
+- Add documentation for the high-level interface API and clean the list of documentation tutorials, making a distinct separation among those using Gammapy as a high-level interface API and those using Gammapy as a Python toolbox.
```

Comment:
Can we add also the python scripts generator (Cf. l293)?

Reply:
@Bultako - If you add things like a Python scripts generator to the task list, then I would suggest to split it in two parts: first what's essential for v1.0, then then a "maybe, nice to have, we're not sure yet" part that will not block v0.14 or v0.15 or v1.0 release. My suggestion would be to list the scripts generator in this second part.

Reply:
+1 on having *required* and *extra* items in the task list.
Since the document seems very oriented towards having an API, I've put the command-line tools items in the extra side.